src/quadtree.ts

```diff
@@ -52,39 +52,90 @@ const TOP_LEFT = 1,
   BOTTOM_LEFT = 3,
   BOTTOM_RIGHT = 4;
 
-interface Boundaries {
+export interface Boundaries {
   x: number;
   y: number;
   width: number;
   height: number;
 }
 
-interface Rectangle {
+export interface Rectangle {
   x1: number;
   y1: number;
   x2: number;
   y2: number;
   height: number;
 }
 
+export interface Vector {
+  x: number;
+  y: number;
+}
+
 /**
  * Geometry helpers.
  */
 
 /**
  * Function returning whether the given rectangle is axis-aligned.
  *
- * @param  {number} x1
- * @param  {number} y1
- * @param  {number} x2
- * @param  {number} y2
+ * @param  {Rectangle} rect
  * @return {boolean}
  */
-function isAxisAligned(x1: number, y1: number, x2: number, y2: number): boolean {
-  return x1 === x2 || y1 === y2;
+export function isRectangleAligned(rect: Rectangle): boolean {
+  return rect.x1 === rect.x2 || rect.y1 === rect.y2;
+}
+
+/**
+ * Function returning the smallest rectangle that contains the given rectangle, and that is aligned with the axis.
+ *
+ * @param {Rectangle} rect
+ * @return {Rectangle}
+ */
+export function getCircumscribedAlignedRectangle(rect: Rectangle): Rectangle {
+  const width = Math.sqrt(Math.pow(rect.x2 - rect.x1, 2) + Math.pow(rect.y2 - rect.y1, 2));
+  const heightVector = {
+    x: ((rect.y1 - rect.y2) * rect.height) / width,
+    y: ((rect.x2 - rect.x1) * rect.height) / width,
+  };
+
+  // Compute all corners:
+  const tl: Vector = { x: rect.x1, y: rect.y1 };
+  const tr: Vector = { x: rect.x2, y: rect.y2 };
+  const bl: Vector = {
+    x: rect.x1 + heightVector.x,
+    y: rect.y1 + heightVector.y,
+  };
+  const br: Vector = {
+    x: rect.x2 + heightVector.x,
+    y: rect.y2 + heightVector.y,
+  };
+
+  const xL = Math.min(tl.x, tr.x, bl.x, br.x);
+  const xR = Math.max(tl.x, tr.x, bl.x, br.x);
+  const yT = Math.min(tl.y, tr.y, bl.y, br.y);
+  const yB = Math.max(tl.y, tr.y, bl.y, br.y);
+
+  return {
+    x1: xL,
+    y1: yT,
+    x2: xR,
+    y2: yT,
+    height: yB - yT,
+  };
 }
 
-function squareCollidesWithQuad(
+/**
+ *
```

Comment:
Strange to have only the names of the params in the jsdoc. Is this some automatic generation? And on a larger level: do we still need jsdoc with typescript now?